app/models/cart.rb

```diff
@@ -0,0 +1,29 @@
+class Cart < ApplicationRecord
+
+
+  belongs_to :user, optional: true
+  has_many :cart_items
+  has_many :items,through: :cart_items
+
+
+  def add_item(item)
+    cart_item=self.cart_items.find_by(item_id:item.id)
+
+    if cart_item
+      cart_item.update(quantity:cart_item[:quantity]+1)
+    else
+      CartItem.create(item_id:item.id,cart_id:self.id,quantity:1)
+    end
+  end
+
+  def remove_item(item)
+    cart_item=self.cart_items.find_by(item_id:item.id)
+    
+    if cart_item[:quantity]===1
+      cart_item.destroy
+    else
+      cart_item.update(quantity:cart_item[:quantity]-1)
+    end
```

Comment:
This logic seem too vague and non logical . Please explain this.

Reply:
If there is just one item of type and you remove it then it will get destroy otherwise if there are more items of type then quantity will be decreased by 1 .

Reply:
changing quantity is different thing and removing cart is different.
**Please follow Single responsibility principle.**

Reply:
cart is not destroying its item existence inside the cart (cart_item) destroys